### PR TITLE
Added webapp-storage-keyvault-msi sample

### PR DESCRIPTION
**Pull request by kaevans**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 301-webapp-storage-keyvault-msi/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# aspnet-mvc-keyvault-storage
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/README.md]: use a friendly title

File: 301-webapp-storage-keyvault-msi/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# aspnet-mvc-keyvault-storage
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F100-blank-template%2Fazuredeploy.json" target="_blank">
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/README.md]: update this to point to your sample

File: 301-webapp-storage-keyvault-msi/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# aspnet-mvc-keyvault-storage
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F100-blank-template%2Fazuredeploy.json" target="_blank">
+<img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F100-blank-template%2Fazuredeploy.json" target="_blank">
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/README.md]: update this to point to your sample

File: 301-webapp-storage-keyvault-msi/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F100-blank-template%2Fazuredeploy.json" target="_blank">
+<img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F100-blank-template%2Fazuredeploy.json" target="_blank">
+<img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.png"/>
+</a>
+
+This template deploys an **Azure Web App** with an Azure AD managed identity that is granted permission to get secrets from **Azure Key Vault**. An **Azure Storage** account is created and its key is stored in the Azure Key Vault. The Azure Web App configuration is updated with the URL to Azure Key Vault and Azure Storage account, and a sample web application is deployed from GitHub.
+
+`Tags: Storage, KeyVault, AppService`
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/README.md]: move tags to the bottom of the readme

File: 301-webapp-storage-keyvault-msi/README.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
++ **sourceControls**: Deploys a sample web application.
+
+#### Key Vault Access Policy
+
+Updates the Azure Key Vault with an access policy allowing the managed identity of the Web App to get secrets from the Key Vault. 
+
+
+## Deployment steps
+
+You can click the "deploy to Azure" button at the beginning of this document. Once deployed, copy the URL for the web application and update the Azure Active Directory application registration with the Reply URL of the web app suffixed with `/signin-oidc`.
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/README.md]: fix casing "Deploy to Azure"

File: 301-webapp-storage-keyvault-msi/scripts/deploy.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/scripts/deploy.sh]: remove this file, scripts need to be deployable using the scripts in the root of the repo

File: 301-webapp-storage-keyvault-msi/README.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+ex. `https://demo.azurewebsites.net/signin-oidc`
+
+Alternatively, **update** the azuredeploy.parameters.json file with the correct parameters, then run the script `deploy.sh` in the `scripts` folder, providing the following parameters:
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/README.md]: remove this section, scripts need to be deployable using the scripts in the root of the repo

File: 301-webapp-storage-keyvault-msi/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+        "skuName": "F1",
+        "skuCapacity": "1",
+        "storageType": "Standard_LRS",
+        "containerName": "demo"
+    },
+    "resources": [
+        {
+            "name": "[parameters('storageAccountName')]",
+            "type": "Microsoft.Storage/storageAccounts",
+            "location": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: 301-webapp-storage-keyvault-msi/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+                {
+                    "apiVersion": "2018-02-01",
+                    "name": "web",
+                    "type": "sourcecontrols",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.Web/Sites', parameters('webAppName'))]",
+                        "[resourceId('Microsoft.Web/Sites/config', parameters('webAppName'),'appsettings')]"
+                    ],
+                    "properties": {
+                        "RepoUrl": "https://github.com/kaevans/aspnet-mvc-keyvault-storage.git",
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/azuredeploy.json]: you cannot deploy from source of a personal repo in these samples... you can add the code here, use webdeploy (and add here) or just remove this resource

File: 301-webapp-storage-keyvault-msi/metadata.json [hunk 10]
```diff
@@ -0,0 +1,9 @@
+{
+   "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+   "type": "QuickStart",
+    "itemDisplayName": "webapp-keyvault-storage-msi",
```
Comment (bmoore-msft) [inline 301-webapp-storage-keyvault-msi/metadata.json]: use a friendly name here

**Comment by kaevans** [comment]
Closing. Doing a demo for Ready. Will make changes and resubmit, but under a different PR.

**Pull request closed without merging**